error checking reorgs. Retrying... Err: object not found

```log
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
2022-01-19T16:16:27Z    ERROR   synchronizer/synchronizer.go:133        error checking reorgs. Retrying... Err: object not found
```

--- a/cmd/register.go
+++ b/cmd/register.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"strings"
 
@@ -11,7 +12,6 @@ import (
 	"github.com/hermeznetwork/hermez-core/state/pgstatestorage"
 	"github.com/hermeznetwork/hermez-core/state/tree"
 	"github.com/iden3/go-iden3-crypto/poseidon"
-	"github.com/jackc/pgx/v4"
 	"github.com/urfave/cli/v2"
 )
 
@@ -62,7 +62,7 @@ func registerSequencer(ctx *cli.Context) error {
 	st := state.NewState(stateCfg, stateDb, tr)
 
 	_, err = st.GetSequencer(ctx.Context, etherman.GetAddress())
-	if err == pgx.ErrNoRows { //If It doesn't exist, register the sequencer
+	if errors.Is(err, state.ErrNotFound) { //If It doesn't exist, register the sequencer
 		tx, err := etherman.RegisterSequencer(url)
 		if err != nil {
 			log.Error("uff no: ", err)

--- a/jsonrpc/chainid.go
+++ b/jsonrpc/chainid.go
@@ -2,10 +2,10 @@ package jsonrpc
 
 import (
 	"context"
+	"errors"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/hermeznetwork/hermez-core/state"
-	"github.com/jackc/pgx/v4"
 )
 
 type chainIDSelector struct {
@@ -28,7 +28,7 @@ func (s *chainIDSelector) getChainID() (uint64, error) {
 		return sequencer.ChainID.Uint64(), nil
 	}
 
-	if err == pgx.ErrNoRows {
+	if errors.Is(err, state.ErrNotFound) {
 		return s.defaultChainID, nil
 	}
 

--- a/jsonrpc/handler.go
+++ b/jsonrpc/handler.go
@@ -50,7 +50,7 @@ func newJSONRpcHandler(e *Eth, n *Net, h *Hez) *Handler {
 // Handle is the function that knows which and how a function should
 // be executed when a JSON RPC request is received
 func (d *Handler) Handle(req Request) Response {
-	log.Debugf("request method %s id %v", req.Method, req.ID)
+	log.Debugf("request method %s id %v params %v", req.Method, req.ID, string(req.Params))
 
 	service, fd, err := d.getFnHandler(req)
 	if err != nil {
@@ -75,7 +75,7 @@ func (d *Handler) Handle(req Request) Response {
 
 	output := fd.fv.Call(inArgs)
 	if err := getError(output[1]); err != nil {
-		log.Errorf("failed to call method %s: %v", req.Method, err)
+		log.Errorf("failed to call method %s: %v. Params: %v", req.Method, err, string(req.Params))
 		return NewRPCResponse(req, nil, newInvalidRequestError(err.Error()))
 	}
 

--- a/synchronizer/synchronizer.go
+++ b/synchronizer/synchronizer.go
@@ -10,7 +10,6 @@ import (
 	"github.com/hermeznetwork/hermez-core/etherman"
 	"github.com/hermeznetwork/hermez-core/log"
 	"github.com/hermeznetwork/hermez-core/state"
-	"github.com/jackc/pgx/v4"
 )
 
 // Synchronizer connects L1 and L2
@@ -261,10 +260,9 @@ func (s *ClientSynchronizer) checkReorg(latestBlock *state.Block) (*state.Block,
 		if (block.Hash() != latestBlock.BlockHash || block.ParentHash() != latestBlock.ParentHash) && latestBlock.BlockNumber > s.genBlockNumber {
 			// Reorg detected. Getting previous block
 			latestBlock, err = s.state.GetBlockByNumber(s.ctx, latestBlock.BlockNumber-1)
-			if err != nil {
-				if err.Error() == pgx.ErrNoRows.Error() {
-					return nil, nil
-				}
+			if errors.Is(err, state.ErrNotFound) {
+				return nil, nil
+			} else if err != nil {
 				return nil, err
 			}
 		} else {

